### PR TITLE
UML-694 Passwords wrapped by HiddenString to prevent leakage

### PR DESCRIPTION
**Pull request by SeemaMenon**
# Purpose
https://opgtransform.atlassian.net/browse/UML-694
Currently we pass around string passwords from handlers to services, these should be wrapped by 
HiddenString as soon as possible to prevent leakage in log files.

Fixes UML-694

## Approach

_Explain how your code addresses the purpose of the change_

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Use a Lasting Power of Attorney service_

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

**Review by johnnolan — DISMISSED** [review]
With my limited PHP skills, the code looks sensible to me

**Review by nickdavis2001 — APPROVED** [review]
lgtm

File: service-api/app/src/App/src/Service/User/UserService.php [hunk 1]
```diff
@@ -85,7 +86,7 @@ public function add(array $data): array
         $activationToken = Base64UrlSafe::encode(random_bytes(32));
         $activationTtl = time() + (60 * 60 * 24);
 
-        $user = $this->usersRepository->add($id, $data['email'], $data['password'], $activationToken, $activationTtl);
+        $user = $this->usersRepository->add($id, $data['email'], new HiddenString($data['password']), $activationToken, $activationTtl);
```
Comment (cooperaj) [inline service-api/app/src/App/src/Service/User/UserService.php:89]: I would argue that the data array should contain a hiddenstring already. It should not be being made in this service class

**Review by cooperaj — CHANGES_REQUESTED** [review]
As per the comment below, we shouldn't be passing passwords from the user input into service layers of the application, the handlers need to wrap passwords before passing them on.

File: service-api/app/src/App/src/Service/User/UserService.php (hunk 1 above)
Comment (SeemaMenon) [reply service-api/app/src/App/src/Service/User/UserService.php]: Made the changes and it works [thought that error'd for me previously]

**Review by cooperaj — APPROVED** [review]
LGTM

**Review by GemTay — APPROVED** [review]

**Pull request merged**